### PR TITLE
feat(HyperlinkButton): lightweight styling for hyperlinkbutton

### PR DESCRIPTION
**Pull request by rajamatt**
﻿GitHub Issue: closes https://github.com/unoplatform/uno.themes-private/issues/7

## PR Type

What kind of change does this PR introduce?

- Feature

## Description

- Continues lightweight styling for HyperlinkButton.

## PR Checklist 
Please check if your PR fulfills the following requirements:

- Commits must be following the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] Tested UWP
- [x] Tested iOS
- [x] Tested Android
- [x] Tested WASM
- [ ] Tested MacOS
- [x] Contains **No** breaking changes
  > If the pull request contains breaking changes, commit message must contain a detailed description of the action to take for the consumer of this library. As explained by the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: src/library/Uno.Themes.WinUI.Markup/Theme.LightWeight/Themes.LightWeightStyles.HyperlinkButton.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+				}
+
+				public static class Secondary
+				{
+					public static class Foreground
+					{
+						[ResourceKeyDefinition(typeof(Brush), "SecondaryHyperlinkButtonForeground")]
+						public static ResourceValue<Brush> Default => new("SecondaryHyperlinkButtonForeground", true);
+					}
+				}
```
Comment (rajamatt) [inline src/library/Uno.Themes.WinUI.Markup/Theme.LightWeight/Themes.LightWeightStyles.HyperlinkButton.cs:96]: @agneszitte @Xiaoy312 not sure about this part, is there a better way (or correct way) for this secondary brush? there's only one: https://github.com/unoplatform/Uno.Themes/blob/66d109c155259d176108a81e9ff1653daeb672a6/src/library/Uno.Material/Styles/Controls/v2/HyperlinkButton.xaml#L35

Comment (Xiaoy312) [reply src/library/Uno.Themes.WinUI.Markup/Theme.LightWeight/Themes.LightWeightStyles.HyperlinkButton.cs]: the 2ndary style is not really implemented fully
just nuke that resource, and inline value on the setter

Comment (agneszitte) [reply src/library/Uno.Themes.WinUI.Markup/Theme.LightWeight/Themes.LightWeightStyles.HyperlinkButton.cs]: I am not following @Xiaoy312, secondary style is the same as the default main style with only a different foreground at the moment in XAML (cc @kazo0)
https://github.com/unoplatform/Uno.Themes/blob/e0e260d6415f95966b95789be4ca79c7e72fb3f7/src/library/Uno.Material/Styles/Controls/v2/HyperlinkButton.xaml#L162

Comment (agneszitte) [reply src/library/Uno.Themes.WinUI.Markup/Theme.LightWeight/Themes.LightWeightStyles.HyperlinkButton.cs]: If we decide to remove the Secondary style it should be done in a separated PR first for the XAML then C# Markup should match

Comment (Xiaoy312) [reply src/library/Uno.Themes.WinUI.Markup/Theme.LightWeight/Themes.LightWeightStyles.HyperlinkButton.cs]: im not saying to nuke the 2ndary style, just its resources and inline the values

Comment (Xiaoy312) [reply src/library/Uno.Themes.WinUI.Markup/Theme.LightWeight/Themes.LightWeightStyles.HyperlinkButton.cs]: lets fast-track this #1092

File: src/library/Uno.Themes.WinUI.Markup/Theme/HyperlinkButton.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+namespace Uno.Themes.Markup
+{
+	public static partial class Theme
+	{
+		public static class HyperlinkButton
+		{
+			public static class Resources
+			{
+				public static class Default
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Theme/HyperlinkButton.cs:13]: I don't think we need this class at this level. It should just be HyperlinkButton.Resources.Foreground.Default/PointerOver/Pressed/etc.

File: src/library/Uno.Themes.WinUI.Markup/Theme.LightWeight/Themes.LightWeightStyles.HyperlinkButton.cs (hunk 1 above)
Comment (Arieldelossantos) [reply src/library/Uno.Themes.WinUI.Markup/Theme.LightWeight/Themes.LightWeightStyles.HyperlinkButton.cs]: fixed

File: src/library/Uno.Themes.WinUI.Markup/Theme/HyperlinkButton.cs (hunk 2 above)
Comment (Arieldelossantos) [reply src/library/Uno.Themes.WinUI.Markup/Theme/HyperlinkButton.cs]: removed

File: src/library/Uno.Themes.WinUI.Markup/Theme/HyperlinkButton.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+					public static ResourceValue<Brush> PointerOver => new("HyperlinkButtonForegroundPointerOver", true);
+
+					[ResourceKeyDefinition(typeof(Brush), "HyperlinkButtonForegroundPressed")]
+					public static ResourceValue<Brush> Pressed => new("HyperlinkButtonForegroundPressed", true);
+
+					[ResourceKeyDefinition(typeof(Brush), "HyperlinkButtonForegroundDisabled")]
+					public static ResourceValue<Brush> Disabled => new("HyperlinkButtonForegroundDisabled", true);
+
+					[ResourceKeyDefinition(typeof(Brush), "SecondaryHyperlinkButtonForeground")]
+					public static ResourceValue<Brush> Secondary => new("SecondaryHyperlinkButtonForeground", true);
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Theme/HyperlinkButton.cs:28]: This resource was removed [here](https://github.com/unoplatform/Uno.Themes/pull/1092/files) so we can remove it here

File: src/library/Uno.Themes.WinUI.Markup/Theme/HyperlinkButton.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+					public static ResourceValue<Brush> PointerOver => new("HyperlinkButtonBorderBrushPointerOver", true);
+
+					[ResourceKeyDefinition(typeof(Brush), "HyperlinkButtonBorderBrushPressed")]
+					public static ResourceValue<Brush> Pressed => new("HyperlinkButtonBorderBrushPressed", true);
+
+					[ResourceKeyDefinition(typeof(Brush), "HyperlinkButtonBorderBrushDisabled")]
+					public static ResourceValue<Brush> Disabled => new("HyperlinkButtonBorderBrushDisabled", true);
+				}
+
+				public static class Typo
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Theme/HyperlinkButton.cs:61]:
```suggestion
				public static class Typography
```

File: src/library/Uno.Themes.WinUI.Markup/Theme/HyperlinkButton.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+				public static ResourceValue<double> Opacity => new("HyperlinkButtonBackgroundOpacity", true);
+
+				[ResourceKeyDefinition(typeof(double), "HyperlinkButtonBackgroundOpacityPointerOver")]
+				public static ResourceValue<double> OpacityPointerOver => new("HyperlinkButtonBackgroundOpacityPointerOver", true);
+
+				[ResourceKeyDefinition(typeof(double), "HyperlinkButtonBackgroundOpacityPressed")]
+				public static ResourceValue<double> OpacityPressed => new("HyperlinkButtonBackgroundOpacityPressed", true);
+
+				[ResourceKeyDefinition(typeof(double), "HyperlinkButtonBackgroundOpacityDisabled")]
+				public static ResourceValue<double> OpacityDisabled => new("HyperlinkButtonBackgroundOpacityDisabled", true);
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Theme/HyperlinkButton.cs:80]: Create an `Opacity` nested class and properties for the different visual state values (with a `Default` for `HyperlinkButtonBackgroundOpacity`)

**Pull request merged**